server/src/main/java/org/apache/iotdb/db/qp/logical/crud/QueryOperator.java

```diff
@@ -337,6 +347,7 @@ protected AlignByDevicePlan generateAlignByDevicePlan(PhysicalGenerator generato
     alignByDevicePlan.setEnableTracing(enableTracing);
 
     alignByDevicePlan.deduplicate(generator);
+    checkWithoutNullColumnValidAlignByDevice(alignByDevicePlan);
```

Comment:
set `withoutNullValidSet` in `AlignByDevicePlan` to null here, friendly for gc.

Reply:
fix